src/pages/api/auth-connect/send-identity-validation-code.ts

```diff
@@ -87,38 +109,32 @@ const handler: Handler = (request, response): Promise<void> => {
           return;
         })
         .catch((error) => {
-          if (
-            error instanceof IdentityAlreadyUsedError ||
-            error instanceof IdentityValueCantBeBlankError
-          ) {
-            span.setDisclosedAttribute(
-              "Identity input exception",
-              error.message,
-            );
-
-            response.statusCode = HttpStatus.BAD_REQUEST;
-            response.json({ errorMessage: error.message });
-            return;
+          span.setDisclosedAttribute("is validation code sent", false);
+
+          if (error instanceof IdentityAlreadyUsedError) {
+            throw webErrorFactory(webErrors.badRequest);
           }
```

Comment:
I was thinking about task `CU-egxugf`. At the moment, Sentry reports exception raised in case of a wrong input, which is an issue. We need to have a way to recognise this error inside the `withAuth` middleware, without telling explicitly that the Identity inputed is already taken.

I can think of two solutions, one expensive, one sub-optimal.

The first one is to add more abstraction to our error's codes, something like what we have in `sparta-api`. This is expensive because we'll need to change all our current web errors, and we all agree that the error's codes in `sparta-api` are probably not the best solution.

The second one would be to add an error content value in the web error, with something like a code. I don't like this one because it would make the algorithm part in `with-auth` obnoxious.

What do you guys think?

Reply:
I guess we could do this in an other PR.